main.tf

```diff
@@ -17,12 +17,12 @@ resource "aws_elasticache_replication_group" "redis" {
 }
 
 resource "aws_elasticache_parameter_group" "redis_parameter_group" {
-  name        = "tf-redis-${var.name}-${data.aws_vpc.vpc.tags["Name"]}"
+  name        = "tf-redis-${var.name}-${replace(format("%.100s", lower(replace("${data.aws_vpc.vpc.tags["Name"]}", "_", "-"))), "/\\s/", "-")}"
```

Comment:
Cool, I think I'd like to make the character limit 255, since thats the [upstream limit from aws.](http://docs.aws.amazon.com/AmazonElastiCache/latest/UserGuide/ParameterGroups.Creating.html) I'm reluctant to place arbitrary limits on resources that differ to much from whats supported by the upstream API. Although I *totally* agree that it isn't desirable to have a name that long.   

You also make a really good point about the use of underscores in the `$var.name` variable. I hadn't considered that!